Feature/gretel opik integration

MAVRICK-1 opened the pull request:
## Details

## Issues

Resolves #2541

## Testing

[Screencast from 2025-06-25 23-59-07.webm](https://github.com/user-attachments/assets/1057cc89-a174-4659-9b3f-fe3e6a0c0cc6)



![Screenshot from 2025-06-25 23-44-21](https://github.com/user-attachments/assets/b0e13d01-0c31-4ef4-839f-913f7d8c9a6e)



/claim #2541

vincentkoc commented:
@MAVRICK-1 Hi Rishi, please give us time to review. And as per the bounty guidelines you need to ask before submitting to avoid one person blocking all the bounties as you have others in-review. Please bear with us

MAVRICK-1 commented:
@vincentkoc sorry 😔, I will keep in mind

vincentkoc commented on apps/opik-documentation/documentation/docs/cookbook/gretel_opik_integration_cookbook.ipynb:
```diff
@@ -0,0 +1,816 @@
+   "source": [
+    "The gretel-qa-dataset dataset can now be viewed in the UI:\n",
+    "\n",
+    "![Gemini Cookbook](https://raw.githubusercontent.com/comet-ml/opik/main/apps/opik-documentation/documentation/fern/img/cookbook/gretel_opik_integration_cookbook.png)"
```
All good, i loaded the notebook but its referencing main.

vincentkoc commented on apps/opik-documentation/documentation/docs/cookbook/gretel_opik_integration_cookbook.ipynb:
```diff
@@ -0,0 +1,816 @@
+    "        # Create or get dataset\n",
+    "        opik_dataset = opik_client.get_or_create_dataset(\n",
+    "            name=dataset_name,\n",
+    "            description=f\"Synthetic Q&A dataset generated using Gretel Navigator ({working_model})\"\n",
```
@MAVRICK-1 please see

vincentkoc commented:
> @vincentkoc done

Thanks ill review

jverre commented:
Hi @MAVRICK-1 
I've been taking a look at the notebook and we will need to make quite a few changes similar to what we did for ADK notebook. The notebook today is very complex which makes it hard to understand how to use Gretel and Opik together.

Let's try to simplify it (for example, the upload to dataset method is ~100 lines of code when it could be just 5 lines - There are other examples of this) and let's also add more text that tells a story of what we are doing in the Notebook.

jverre reviewed (changes requested):
See comment above

MAVRICK-1 commented:
@jverre 

## Updated Gretel Integration: Replaced Deprecated Methods

**Issue**: The old Gretel API methods were deprecated and no longer working, causing integration failures.

**Solution**: I implemented new methods using the current Gretel APIs:

1. **Gretel Data Designer** - https://docs.gretel.ai/create-synthetic-data/gretel-data-designer
   - For creating custom Q&A datasets from scratch
   - Uses modular column design with LLM-generated content

2. **Safe Synthetics** - https://docs.gretel.ai/create-synthetic-data/safe-synthetics  
   - For privacy-safe synthesis of existing datasets
   - Includes PII protection and compliance features

**Result**: Successfully created a working Gretel-to-Opik integration that generates synthetic Q&A datasets and uploads them to Opik for model evaluation.

The cookbook now uses current, supported APIs ensuring long-term reliability.
